File: app/src/main/java/com/seebaldtart/projectmusicplayer/MainActivity.java
package com.seebaldtart.projectmusicplayer;
import android.content.Context;
import android.content.Intent;
import android.media.AudioManager;
import android.media.MediaPlayer;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ImageButton;
import android.widget.ListView;
import android.widget.SeekBar;
import android.widget.TextView;
import java.text.DecimalFormat;
import java.text.NumberFormat;
import java.util.ArrayList;
public class MainActivity extends AppCompatActivity {
    static int currentPosition = 0;        // Current Playback Position
    static int cycle = 0;
    static ArrayList<String> songStringList = new ArrayList<>();
    static ArrayList<SongObject> songList = new ArrayList<>();
    static MediaPlayer media;
    static AudioManager audioManager;
    static java.lang.reflect.Field[] fields;
    static Context activityContext;
    static SongObject currentSong;
    static boolean focusGranted;
    static boolean wasPlaying;
    ImageButton playButton;
    ImageButton nextButton;
    ImageButton previousButton;
    static TextView selectedTrack;
    TextView currentDurationText;
    TextView totalDurationText;
    SeekBar seekBar;
    static Runnable runnable;
    static Handler handler;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        activityContext = getApplicationContext();
        playButton = findViewById(R.id.play);
        nextButton = findViewById(R.id.next);
        previousButton = findViewById(R.id.previous);
        selectedTrack = findViewById(R.id.current_song_text);
        currentDurationText = findViewById(R.id.current_duration);
        totalDurationText = findViewById(R.id.total_duration);
        seekBar = findViewById(R.id.seekbar);
        audioManager = (AudioManager) getApplicationContext().getSystemService(Context.AUDIO_SERVICE);
        handler = new Handler();
        playButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (isFocusGranted()) {
                    if (media == null) {        // Nothing has been played...
                        playMediaFromPosition(0);
                        playButton.setImageResource(R.drawable.baseline_pause_white_24);
                    } else {        // If media has already been defined (if something has been played...)
                        if (media.isPlaying()) {     // If media IS currently playing...
                            media.pause();
                            playButton.setImageResource(R.drawable.baseline_play_arrow_white_24);
                        } else {
                            if (media != null) {
                                media.start();
                                playButton.setImageResource(R.drawable.baseline_pause_white_24);
                            }
                        }
                    }
                }
            }
        });
        nextButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (media.isPlaying()) {
                    media.stop();
                    media.release();
                    playMediaFromPosition(changeTrackTo(currentPosition, 1));
                    playButton.setImageResource(R.drawable.baseline_pause_white_24);
                } else {
                    media.stop();
                    media.release();
                    media = MediaPlayer.create(MainActivity.this, changeTrackTo(1));
                    playButton.setImageResource(R.drawable.baseline_play_arrow_white_24);
                }
            }
        });
        previousButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int currentDurationPosition = media.getCurrentPosition();
                if (media != null) {
                    if (media.isPlaying()) {
                        media.pause();
                        playButton.setImageResource(R.drawable.baseline_play_arrow_white_24);
                        if (currentDurationPosition < 3000) {       // change track
                            media.stop();
                            media.release();
                            media = MediaPlayer.create(MainActivity.this, changeTrackTo(-1));
                        } else {        // restart track
                            media.seekTo(0);
                            media.start();
                            playButton.setImageResource(R.drawable.baseline_pause_white_24);
                        }
                        media.start();
                        playButton.setImageResource(R.drawable.baseline_pause_white_24);
                    } else {
                        if (currentDurationPosition < 3000) {       // change track
                            media.stop();
                            media.release();
                            media = MediaPlayer.create(MainActivity.this, changeTrackTo(-1));
                        } else {        // restart track
                            media.seekTo(0);
                        }
                    }
                }
            }
        });
        fields = R.raw.class.getFields();
        for (int count = 0; count < fields.length; count++) {
            songList.add(new SongObject(fields[count]));
            songStringList.add(fields[count].getName());
        }
        AudioAdapter adapter = new AudioAdapter(this, songList);
        ListView listView = findViewById(R.id.item_list);
        listView.setAdapter(adapter);
        if (media == null) {
            media = MediaPlayer.create(MainActivity.this, getMediaAtPosition(0));
            media.setOnCompletionListener(mCompletionListener);
        }
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                if (isFocusGranted()) {
                    if (media != null) {
                        media.release();
                        playButton.setImageResource(R.drawable.baseline_play_arrow_white_24);
                    }
                    playMediaFromPosition(position);
                    playButton.setImageResource(R.drawable.baseline_pause_white_24);
                }
            }
        });
        selectedTrack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, SelectedSong.class);
                startActivity(intent);
            }
        });
        initializeSeekbar();
        playCycle();
    }
    public void playNextSong() {
        media.stop();
        media.release();
        playMediaFromPosition(changeTrackTo(currentPosition, 1));
        playButton.setImageResource(R.drawable.baseline_pause_white_24);
    }
    public void repeatCurrentSong() {
        media.pause();
        media.seekTo(0);
        media.start();
        playButton.setImageResource(R.drawable.baseline_pause_white_24);
    }
    MediaPlayer.OnCompletionListener mCompletionListener = new MediaPlayer.OnCompletionListener() {
        @Override
        public void onCompletion(MediaPlayer mp) {
            int capOff = songList.size() - 1;
            if (cycle == 0) {           // Repeat List of songs...
                playNextSong();
            } else {            // Repeat current song only...
                repeatCurrentSong();
            }
        }
    };
    static AudioManager.OnAudioFocusChangeListener mFocusChangeListener = new AudioManager.OnAudioFocusChangeListener() {
        @Override
        public void onAudioFocusChange(int focusChange) {
            if (focusChange == AudioManager.AUDIOFOCUS_LOSS) {
                releaseMediaPlayer();
            } else if (focusChange == AudioManager.AUDIOFOCUS_LOSS_TRANSIENT || focusChange == AudioManager.AUDIOFOCUS_LOSS_TRANSIENT_CAN_DUCK) {
                if (media.isPlaying()) {
                    media.pause();
                    wasPlaying = true;
                }
            } else if (focusChange == AudioManager.AUDIOFOCUS_GAIN) {
                if (wasPlaying) {
                    media.start();
                }
            }
        }
    };
    public boolean isFocusGranted() {
        int result = audioManager.requestAudioFocus(mFocusChangeListener, audioManager.STREAM_MUSIC, AudioManager.AUDIOFOCUS_GAIN_TRANSIENT);
        if (result == AudioManager.AUDIOFOCUS_REQUEST_GRANTED) {
            focusGranted = true;
            return true;
        }
        focusGranted = false;
        return false;
    }
    public static void releaseMediaPlayer () {
        if (media != null) {
            media.release();
            media = null;
            audioManager.abandonAudioFocus(mFocusChangeListener);
            currentPosition = 0;
        }
    }
    public void playMediaFromPosition(int pos) {
        media = MediaPlayer.create(MainActivity.this, getMediaAtPosition(pos));
        media.start();
    }
    public int getMediaAtPosition(int pos) {
        currentPosition = pos;
        setCurrentSong(pos);
        int resID = getResources().getIdentifier(songStringList.get(pos), "raw", getPackageName());
        return resID;
    }
    public int changeTrackTo(int value) {
        if (media != null) {
            int nextPos = currentPosition + value;
            if (nextPos < 0) {     // Out of bounds: reverting to bottom of list
                currentPosition = songList.size() - 1;
            }
            else if (nextPos >= 0 && nextPos < songList.size()){
                currentPosition = nextPos;
            }
            else if (nextPos >= songList.size()) {     // Out of bounds: reverting back to top of list
                currentPosition = 0;
            }
        } else {
            currentPosition = 0;
        }
        currentSong = songList.get(currentPosition);
        selectedTrack.setText(getSelectedTrackText(currentSong));
        int resID = getResources().getIdentifier(songStringList.get(currentPosition), "raw", getPackageName());
        return resID;
    }
    public int changeTrackTo(int sv, int increment) {
        int startingValue = sv;
        if (media != null) {
            int nextPos = startingValue + increment;
            if (nextPos < 0) {     // Out of bounds: reverting to bottom of list
                currentPosition = songList.size() - 1;
                startingValue = startingValue -1;
            }
            else if (nextPos >= 0 && nextPos < songList.size()){
                currentPosition = nextPos;
                startingValue = nextPos;
            }
            else if (nextPos >= songList.size()) {     // Out of bounds: reverting back to top of list
                currentPosition = 0;
                startingValue = 0;
            }
        } else {
            currentPosition = 0;
            startingValue = 0;
        }
        currentSong = songList.get(currentPosition);
        selectedTrack.setText(getSelectedTrackText(currentSong));
        return startingValue;
    }
    public void setCurrentSong(int pos) {
        currentSong = songList.get(pos);
        selectedTrack.setText(getSelectedTrackText(currentSong));
    }
    public String getSelectedTrackText(SongObject song) {
        String songTitle = song.getSongTitle();
        String artistName = song.getArtistName();
        return songTitle + " - " + artistName;
    }
    public String getMediaTime(int time) {
        int min = (time / 1000) / 60;
        int sec = (time / 1000) % 60;
        NumberFormat formatter = new DecimalFormat("00");
        String minutes = "";
        if (min < 10) {
            minutes = String.format("%02d", min);
        } else {
            minutes = String.format("%01d", min);
        }
        String seconds = formatter.format(sec);
        return minutes + ":" + seconds;
    }
    private void initializeSeekbar() {
        seekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            int currentProgress;
            boolean mediaWasPlaying = false;
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean inputFromUser) {
                seekBar.setProgress(progress);
                currentProgress = progress;
                if (inputFromUser) {
                    currentDurationText.setText(getMediaTime(progress));
                }
                currentDurationText.setText(getMediaTime(progress));
            }
            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
                if (media.isPlaying()) {
                    media.pause();
                    mediaWasPlaying = true;
                }
                currentDurationText.setText(getMediaTime(currentProgress));
            }
            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
                currentDurationText.setText(getMediaTime(currentProgress));
                if (mediaWasPlaying) {
                    media.seekTo(currentProgress);
                    media.start();
                    mediaWasPlaying = false;
                }
            }
        });
    }
    private void playCycle() {              // Retrieved from: https://www.youtube.com/watch?v=HB3DoZh1QWU
        if (media != null) {
            seekBar.setProgress(media.getCurrentPosition());
            seekBar.setMax(media.getDuration());
            totalDurationText.setText(getMediaTime(media.getDuration()));
            media.setOnCompletionListener(mCompletionListener);
            runnable = new Runnable() {
                @Override
                public void run() {
                    playCycle();
                }
            };
            handler.postDelayed(runnable, 100);
        }
    }
    @Override
    protected void onPause() {
        super.onPause();
    }
    @Override
    protected void onResume() {
        super.onResume();
        if (media != null) {
            if (!media.isPlaying()) {
                wasPlaying = false;
            } else {
                wasPlaying = true;
            }
        }
        initializeSeekbar();
    }
    @Override
    protected void onStop () {
        super.onStop();
    }
    @Override
    protected void onDestroy() {
        super.onDestroy();
        releaseMediaPlayer();
        handler.removeCallbacks(runnable);          // Retrieved from: https://www.youtube.com/watch?v=HB3DoZh1QWU
    }
}

File: app/src/main/java/com/seebaldtart/projectmusicplayer/ui/activities/MainActivity.kt
package com.seebaldtart.projectmusicplayer.ui.activities

import android.content.Context
import android.media.AudioManager
import android.media.MediaPlayer
import android.media.MediaPlayer.OnCompletionListener
import android.os.Bundle
import android.os.Handler
import android.widget.ImageButton
import android.widget.SeekBar
import android.widget.SeekBar.OnSeekBarChangeListener
import android.widget.TextView
import androidx.activity.compose.setContent
import androidx.appcompat.app.AppCompatActivity
import androidx.hilt.navigation.compose.hiltViewModel
import com.seebaldtart.projectmusicplayer.R
import com.seebaldtart.projectmusicplayer.models.SongObject
import com.seebaldtart.projectmusicplayer.viewmodels.MusicPlayerStateViewModel
import dagger.hilt.android.AndroidEntryPoint
import java.lang.reflect.Field
import java.text.DecimalFormat
import java.text.NumberFormat
import javax.inject.Inject

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    private var playButton: ImageButton? = null
    private var nextButton: ImageButton? = null
    private var previousButton: ImageButton? = null
    private var currentDurationText: TextView? = null
    private var totalDurationText: TextView? = null
    private var seekBar: SeekBar? = null

    private var currentPosition: Int = 0 // Current Playback Position
    private var cycle: Int = 0
    private var songStringList: ArrayList<String> = ArrayList()
    private var songList: ArrayList<SongObject> = ArrayList()
    private var media: MediaPlayer? = null
    private var audioManager: AudioManager? = null
    private var fields: Array<Field>? = null
    private var activityContext: Context? = null
    private var currentSong: SongObject? = null
    private var focusGranted: Boolean = false
    private var wasPlaying: Boolean = false
    private var selectedTrack: TextView? = null
    private var runnable: Runnable? = null
    private var handler: Handler? = null

    lateinit var viewModel: MusicPlayerStateViewModel

    var mFocusChangeListener: AudioManager.OnAudioFocusChangeListener =
        AudioManager.OnAudioFocusChangeListener { focusChange ->
            if (focusChange == AudioManager.AUDIOFOCUS_LOSS) {
                releaseMediaPlayer()
            } else if (focusChange == AudioManager.AUDIOFOCUS_LOSS_TRANSIENT || focusChange == AudioManager.AUDIOFOCUS_LOSS_TRANSIENT_CAN_DUCK) {
                if (media!!.isPlaying) {
                    media!!.pause()
                    wasPlaying = true
                }
            } else if (focusChange == AudioManager.AUDIOFOCUS_GAIN) {
                if (wasPlaying) {
                    media!!.start()
                }
            }
        }

    private fun releaseMediaPlayer() {
        if (media != null) {
            media!!.release()
            media = null
            audioManager!!.abandonAudioFocus(mFocusChangeListener)
            currentPosition = 0
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        activityContext = applicationContext
        playButton = findViewById(R.id.play) as? ImageButton
        nextButton = findViewById(R.id.next) as? ImageButton
        previousButton = findViewById(R.id.previous) as? ImageButton
        selectedTrack = findViewById(R.id.current_song_text)
        currentDurationText = findViewById(R.id.current_duration)
        totalDurationText = findViewById(R.id.total_duration)
        seekBar = findViewById(R.id.seekbar)
        audioManager = applicationContext.getSystemService(AUDIO_SERVICE) as AudioManager
        handler = Handler()

        setContent {
            viewModel = hiltViewModel<MusicPlayerStateViewModel>()
        }

//        playButton.setOnClickListener(View.OnClickListener { v: View? ->
//            if (isFocusGranted) {
//                if (media == null) {        // Nothing has been played...
//                    playMediaFromPosition(0)
//                    playButton.setImageResource(R.drawable.baseline_pause_white_24)
//                } else {        // If media has already been defined (if something has been played...)
//                    if (media!!.isPlaying) {     // If media IS currently playing...
//                        media!!.pause()
//                        playButton.setImageResource(R.drawable.baseline_play_arrow_white_24)
//                    } else {
//                        if (media != null) {
//                            media!!.start()
//                            playButton.setImageResource(R.drawable.baseline_pause_white_24)
//                        }
//                    }
//                }
//            }
//        })
//        nextButton.setOnClickListener(View.OnClickListener { v: View? ->
//            if (media!!.isPlaying) {
//                media!!.stop()
//                media!!.release()
//                playMediaFromPosition(changeTrackTo(currentPosition, 1))
//                playButton.setImageResource(R.drawable.baseline_pause_white_24)
//            } else {
//                media!!.stop()
//                media!!.release()
//                media = MediaPlayer.create(this@MainActivity, changeTrackTo(1))
//                playButton.setImageResource(R.drawable.baseline_play_arrow_white_24)
//            }
//        })
//        previousButton.setOnClickListener(View.OnClickListener { v: View? ->
//            val currentDurationPosition = media!!.currentPosition
//            if (media != null) {
//                if (media!!.isPlaying) {
//                    media!!.pause()
//                    playButton.setImageResource(R.drawable.baseline_play_arrow_white_24)
//                    if (currentDurationPosition < 3000) {       // change track
//                        media!!.stop()
//                        media!!.release()
//                        media = MediaPlayer.create(this@MainActivity, changeTrackTo(-1))
//                    } else {        // restart track
//                        media!!.seekTo(0)
//                        media!!.start()
//                        playButton.setImageResource(R.drawable.baseline_pause_white_24)
//                    }
//                    media!!.start()
//                    playButton.setImageResource(R.drawable.baseline_pause_white_24)
//                } else {
//                    if (currentDurationPosition < 3000) {       // change track
//                        media!!.stop()
//                        media!!.release()
//                        media = MediaPlayer.create(this@MainActivity, changeTrackTo(-1))
//                    } else {        // restart track
//                        media!!.seekTo(0)
//                    }
//                }
//            }
//        })
//        fields = raw::class.java.fields
//        for (count in fields.indices) {
//            songList.add(SongObject(fields[count]))
//            songStringList.add(fields[count].name)
//        }
//        val adapter =
//            AudioAdapter(
//                this,
//                songList
//            )
//        val listView = findViewById<ListView>(R.id.item_list)
//        listView.adapter = adapter
//        if (media == null) {
//            media = MediaPlayer.create(this@MainActivity, getMediaAtPosition(0))
//            media.setOnCompletionListener(mCompletionListener)
//        }
//        listView.onItemClickListener =
//            OnItemClickListener { parent: AdapterView<*>?, view: View?, position: Int, id: Long ->
//                if (isFocusGranted) {
//                    if (media != null) {
//                        media!!.release()
//                        playButton.setImageResource(R.drawable.baseline_play_arrow_white_24)
//                    }
//                    playMediaFromPosition(position)
//                    playButton.setImageResource(R.drawable.baseline_pause_white_24)
//                }
//            }
//        selectedTrack.setOnClickListener(View.OnClickListener { v: View? ->
//            val intent = Intent(this@MainActivity, SelectedSongActivity::class.java)
//            startActivity(intent)
//        })
        initializeSeekbar()
        playCycle()
    }

    fun playNextSong() {
        media!!.stop()
        media!!.release()
        playMediaFromPosition(changeTrackTo(currentPosition, 1))
        playButton!!.setImageResource(R.drawable.baseline_pause_white_24)
    }

    fun repeatCurrentSong() {
        media!!.pause()
        media!!.seekTo(0)
        media!!.start()
        playButton!!.setImageResource(R.drawable.baseline_pause_white_24)
    }

    var mCompletionListener: OnCompletionListener = OnCompletionListener {
        val capOff = songList.size - 1
        if (cycle == 0) {           // Repeat List of songs...
            playNextSong()
        } else {            // Repeat current song only...
            repeatCurrentSong()
        }
    }
    val isFocusGranted: Boolean
        get() {
            val result = audioManager!!.requestAudioFocus(
                mFocusChangeListener,
                AudioManager.STREAM_MUSIC,
                AudioManager.AUDIOFOCUS_GAIN_TRANSIENT
            )
            if (result == AudioManager.AUDIOFOCUS_REQUEST_GRANTED) {
                focusGranted = true
                return true
            }
            focusGranted = false
            return false
        }

    fun playMediaFromPosition(pos: Int) {
//        media = MediaPlayer.create(this@MainActivity, getMediaAtPosition(pos))
//        media.start()
    }

    fun getMediaAtPosition(pos: Int): Int {
        currentPosition = pos
        setCurrentSong(pos)
        val resID = resources.getIdentifier(songStringList[pos], "raw", packageName)
        return resID
    }

    fun changeTrackTo(value: Int): Int {
        if (media != null) {
            val nextPos = currentPosition + value
            if (nextPos < 0) {     // Out of bounds: reverting to bottom of list
                currentPosition = songList.size - 1
            } else if (nextPos >= 0 && nextPos < songList.size) {
                currentPosition = nextPos
            } else if (nextPos >= songList.size) {     // Out of bounds: reverting back to top of list
                currentPosition = 0
            }
        } else {
            currentPosition = 0
        }
        currentSong = songList[currentPosition]
        selectedTrack!!.text =
            getSelectedTrackText(currentSong)
        val resID = resources.getIdentifier(songStringList[currentPosition], "raw", packageName)
        return resID
    }

    fun changeTrackTo(sv: Int, increment: Int): Int {
        var startingValue = sv
        if (media != null) {
            val nextPos = startingValue + increment
            if (nextPos < 0) {     // Out of bounds: reverting to bottom of list
                currentPosition = songList.size - 1
                startingValue = startingValue - 1
            } else if (nextPos >= 0 && nextPos < songList.size) {
                currentPosition = nextPos
                startingValue = nextPos
            } else if (nextPos >= songList.size) {     // Out of bounds: reverting back to top of list
                currentPosition = 0
                startingValue = 0
            }
        } else {
            currentPosition = 0
            startingValue = 0
        }
        currentSong = songList[currentPosition]
        selectedTrack!!.text =
            getSelectedTrackText(currentSong)
        return startingValue
    }

    fun setCurrentSong(pos: Int) {
        currentSong = songList[pos]
        selectedTrack!!.text =
            getSelectedTrackText(currentSong)
    }

    fun getSelectedTrackText(song: SongObject?): String {
        val songTitle = song!!.songTitle
        val artistName = song.artistName
        return "$songTitle - $artistName"
    }

    fun getMediaTime(time: Int): String {
        val min = (time / 1000) / 60
        val sec = (time / 1000) % 60
        val formatter: NumberFormat = DecimalFormat("00")
        var minutes = ""
        minutes = if (min < 10) {
            String.format("%02d", min)
        } else {
            String.format("%01d", min)
        }
        val seconds = formatter.format(sec.toLong())
        return "$minutes:$seconds"
    }

    private fun initializeSeekbar() {
        seekBar!!.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            var currentProgress: Int = 0
            var mediaWasPlaying: Boolean = false

            override fun onProgressChanged(
                seekBar: SeekBar,
                progress: Int,
                inputFromUser: Boolean
            ) {
                seekBar.progress = progress
                currentProgress = progress
                if (inputFromUser) {
                    currentDurationText!!.text = getMediaTime(progress)
                }
                currentDurationText!!.text = getMediaTime(progress)
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {
                if (media!!.isPlaying) {
                    media!!.pause()
                    mediaWasPlaying = true
                }
                currentDurationText!!.text = getMediaTime(currentProgress)
            }

            override fun onStopTrackingTouch(seekBar: SeekBar) {
                currentDurationText!!.text = getMediaTime(currentProgress)
                if (mediaWasPlaying) {
                    media!!.seekTo(currentProgress)
                    media!!.start()
                    mediaWasPlaying = false
                }
            }
        })
    }

    private fun playCycle() {
//        if (media != null) {
//            seekBar!!.progress = media!!.currentPosition
//            seekBar!!.max = media!!.duration
//            totalDurationText!!.text = getMediaTime(media!!.duration)
//            media!!.setOnCompletionListener(mCompletionListener)
//            runnable = Runnable { playCycle() }
//            handler!!.postDelayed(runnable, 100)
//        }
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onResume() {
        super.onResume()
        if (media != null) {
            if (!media!!.isPlaying) {
                wasPlaying = false
            } else {
                wasPlaying = true
            }
        }
        initializeSeekbar()
    }

    override fun onStop() {
        super.onStop()
    }

    override fun onDestroy() {
        super.onDestroy()
        releaseMediaPlayer()
        handler!!.removeCallbacks(runnable!!)
    }

}